### PR TITLE
Draft: Refactor doca

### PR DESCRIPTION
**Pull request by Ralfons-06**
- Add Doca Domain Bounding Phase
- Refactoring of Doca  
- PR includes Bracnches [modes-doca](https://github.com/fjoniyz/ganges/tree/modes-doca) and [doca-domain-bounding](https://github.com/fjoniyz/ganges/tree/doca-domain-bounding)

### CODE REVIEW

File: src/main/java/com/ganges/lib/castleguard/CastleGuard.java
Comment (nomorehumor) [inline src/main/java/com/ganges/lib/castleguard/CastleGuard.java:1]: only indentation fixes, isn't it?

File: src/main/resources/doca.properties [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #Fields that need to be anonymized in doca
-doca_fields=kwh
+doca_fields=Seconds_EnergyConsumption
```
Comment (nomorehumor) [inline src/main/resources/doca.properties:2]: please rebase on current master and test again, the field name that you use here is from an outdated dataset

Comment (nomorehumor) [reply src/main/resources/doca.properties:2]: (test with current dataset)

**Review by nomorehumor — CHANGES_REQUESTED** [review]

File: src/main/java/anoniks/Pipe.java [hunk 2]
```diff
@@ -137,7 +137,9 @@ private static JsonNode processing(JsonNode message, DataRepository dataReposito
     List<AnonymizationItem> output = algorithm.anonymize(contextValues);
 
     ArrayNode outputMessage = getJsonFromItems(output);
-    System.out.println("OUTPUT " + outputMessage);
+    if (!outputMessage.isEmpty()) {
+      System.out.println("OUTPUT " + outputMessage);
+    }
```
[the remark this answers is not in the document]
Comment (Ralfons-06) [reply src/main/java/anoniks/Pipe.java:142]: No, i just changed it so the terminal wouldn't be so clutered. Will change it back.

File: src/main/resources/doca.properties (hunk 1 above)
Comment (Ralfons-06) [reply src/main/resources/doca.properties]: Tested. Works with the current dataset

**Review by nomorehumor — APPROVED** [review]

**Pull request merged**